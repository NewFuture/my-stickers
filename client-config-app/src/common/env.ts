/**
 * 图片个数限制
 */
export const MAX_NUM = 150;

/**
 * Enable / disable the tenant features.
 */
export const ENABLE_TENANT = true;

/**
 * API 地址
 */
export const BASE_URL = process.env.REACT_APP_API_ROOT || "/api/";

/**
 * AAD Client ID
 */
export const AAD_ID = process.env.REACT_APP_AAD_ID;

/**
 * Query Params when APP start
 */
export const INIT_QUERY = new URLSearchParams(window.location.search);
