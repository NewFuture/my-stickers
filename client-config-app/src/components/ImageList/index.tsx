import ImageList from "./ImageList";

export default ImageList;
