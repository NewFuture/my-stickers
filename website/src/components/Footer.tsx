import { Flex, Button, Divider } from "@fluentui/react-northstar";
import React from "react";
import { useTranslation } from "react-i18next";
import { Link } from "react-router-dom";
import { Common } from "../locales";

const Footer: React.FC = () => {
    const { t } = useTranslation();
    return (
        <footer className="Footer">
            <Divider />
            <Flex vAlign="center" hAlign="center">
                <Button to="/" as={Link} text secondary size="small" color="info">
                    {t(Common.homeTitle)}
                </Button>
                <Button
                    as="a"
                    href="https://github.com/NewFuture/custom-stickers-teams-extension/issues/new"
                    text
                    secondary
                    size="small"
                    target="_blank"
                    rel="noopener noreferrer"
                    color="info"
                >
                    {t(Common.feedback)}
                </Button>
                <Button
                    as="a"
                    href="https://github.com/NewFuture/custom-stickers-teams-extension"
                    text
                    secondary
                    size="small"
                    target="_blank"
                    rel="noopener noreferrer"
                    color="info"
                >
                    Github
                </Button>
                <Button to="/privacy.html" as={Link} text secondary size="small" color="info">
                    {t(Common.privacyTitle)}
                </Button>
                <Button to="/terms.html" as={Link} text secondary size="small" color="info">
                    {t(Common.termsTitle)}
                </Button>
            </Flex>
        </footer>
    );
};

export default Footer;
