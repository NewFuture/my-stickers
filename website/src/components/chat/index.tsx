import ChatWithPopover from "./ChatWithPopover";
import { useTranslation } from "react-i18next";

import { PrototypeComponent, PrototypeSection } from "../Prototype";
import { TransKeys } from "../../locales";

export default function ChatMessage(props: any) {
    const { t } = useTranslation();
    return (
        <PrototypeSection title={t(TransKeys.protoMsgExtTitle)}>
            <PrototypeComponent
                title={t(TransKeys.protoMsgExtSubTitle)}
                // description="Click save stciker in the more action menu list can save it!"
            >
                <ChatWithPopover />
            </PrototypeComponent>
        </PrototypeSection>
    );
}
