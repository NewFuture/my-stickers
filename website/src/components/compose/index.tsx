import React from "react";
import { useTranslation } from "react-i18next";

import { PrototypeComponent, PrototypeSection } from "../Prototype";
import { TransKeys } from "../../locales";
import { ComposeChat } from "./ComposeChat";

export default function Compose(props: React.PropsWithChildren<{}>) {
    const { t } = useTranslation();
    return (
        <PrototypeSection title={t(TransKeys.protoComposeTitle)}>
            <PrototypeComponent
                title={t(TransKeys.protoComposeSubTitle)}
                // description="Click save stciker in the more action menu list can save it!"
            >
                <ComposeChat />
            </PrototypeComponent>
        </PrototypeSection>
    );
}
