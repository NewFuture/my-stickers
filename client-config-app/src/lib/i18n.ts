import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import { INIT_QUERY } from "./env";
import zhCN from "../locales/zh-cn.json";
import en from "../locales/en.json";

const lng = INIT_QUERY.get("lng") || navigator.language || navigator.languages?.[0];

i18n
    // load translation using xhr -> see /public/locales
    // learn more: https://github.com/i18next/i18next-xhr-backend
    //   .use(Backend)
    // pass the i18n instance to react-i18next.
    .use(initReactI18next)
    // init i18next
    // for all options read: https://www.i18next.com/overview/configuration-options
    .init({
        debug: process.env.NODE_ENV === "development",
        lng,
        fallbackLng: "en",
        lowerCaseLng: true,
        interpolation: {
            escapeValue: false, // not needed for react as it escapes by default
        },
        resources: {
            en: { translation: en },
            "zh-cn": { translation: zhCN },
        },
    });

export default i18n;
