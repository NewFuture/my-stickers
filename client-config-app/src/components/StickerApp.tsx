import { Divider, makeStyles } from "@fluentui/react-components";
import { useState } from "react";
import type { UserType } from "../model/sticker";
import Header from "./Header/Header";
import { TenantImageList } from "./ImageList/TenantImageList";
import { UserImageList } from "./ImageList/UserImageList";

const useStyles = makeStyles({
    content: {
        minHeight: "calc(100vh - 64px)",
        "&>.fui-Spinner": {
            minHeight: "calc(100vh - 64px)",
        },
    },
});

export function StickerApp(): JSX.Element {
    const styles = useStyles();
    const [currentRadio, setCurrentRadio] = useState<UserType>("user");
    const isTenant = currentRadio === "company";
    return (
        <>
            <Header type={currentRadio} onRadioChange={setCurrentRadio} />
            <Divider />
            <main className={styles.content}>{isTenant ? <TenantImageList /> : <UserImageList />}</main>
        </>
    );
}
