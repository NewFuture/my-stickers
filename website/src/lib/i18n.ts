import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import LanguageDetector from "i18next-browser-languagedetector";

import * as zh from "../locales/zh";
import * as en from "../locales/en";

// not like to use this?
// have a look at the Quick start guide
// for passing in lng and translations on init

i18n
    // load translation using xhr -> see /public/locales
    // learn more: https://github.com/i18next/i18next-xhr-backend
    //   .use(Backend)
    // detect user language
    // learn more: https://github.com/i18next/i18next-browser-languageDetector
    .use(LanguageDetector)
    // pass the i18n instance to react-i18next.
    .use(initReactI18next)
    // init i18next
    // for all options read: https://www.i18next.com/overview/configuration-options
    .init({
        fallbackLng: "en",
        debug: process.env.NODE_ENV === "development",
        resources: {
            en,
            zh,
            "zh-CN": zh,
        },
        // react i18next special options (optional)
        // override if needed - omit if ok with defaults
        /*
        react: {
          bindI18n: 'languageChanged',
          bindI18nStore: '',
          transEmptyNodeValue: '',
          transSupportBasicHtmlNodes: true,
          transKeepBasicHtmlNodesFor: ['br', 'strong', 'i'],
          useSuspense: true,
        }
        */
    });

export default i18n;
