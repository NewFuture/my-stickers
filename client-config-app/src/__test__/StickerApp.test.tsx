import { create } from "react-test-renderer";
import { FluentProvider, teamsLightTheme } from "@fluentui/react-components";
import { StickerApp } from "../components/StickerApp";
import "../common/i18n";

test("StickerApp Snapshot", () => {
    const tree = create(
        <FluentProvider theme={teamsLightTheme}>
            <StickerApp />
        </FluentProvider>,
    ); // 渲染
    expect(tree).toMatchSnapshot();
});
