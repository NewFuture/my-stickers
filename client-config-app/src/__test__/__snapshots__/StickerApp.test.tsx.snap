// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickerApp Snapshot 1`] = `
<div
  className="fui-FluentProvider fui-FluentProviderr0 ___jdtuxv0_17k0bs4 f19n0e5 fxugw4r f1o700av fk6fouc fkhj508 figsok6 f1i3iumi"
  dir="ltr"
>
  <header
    className="___179c093_1r5nvjz f22iagw f122n59 fni485r f16muhyy fiq88j9 fll10am f1y9r80w f15tla4x"
  >
    <h1
      className="fui-Text ___imzx740_1gseor6 fk6fouc f1x0m3f5 fb86gi6 flh3ekv fpgzoln f1w7gpdv f6juhto f1mtd64y f1y7q3j9 f2jf649"
    >
      My Favorite Stickers
    </h1>
    <div
      aria-haspopup="menu"
      className="fui-SplitButton ___avynhp0_115hgeb ftuwxu6 fsxf2b5 f10pi13n fmrv4ls f14uur2j fye5tvs f1e8brtx fxp12j1 fr7y8no f1hvqvul"
      id="menur1"
      onClick={[Function]}
      onContextMenu={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      role="button"
      tabIndex={0}
      title="stickers scope"
    >
      <button
        className="fui-Button rsawnvh fui-SplitButton__primaryActionButton ___mrobob0_upoub20 ftqa4ok f2hkw1w f1x37qnr fn4c73s"
        disabled={false}
        id="splitButton-r2__primaryActionButton"
        onClick={[Function]}
        type="button"
      >
        <span
          className="fui-Button__icon rywnvv2 ___963sj20_1qpm2bl f1nizpg2"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9 2a4 4 0 1 0 0 8 4 4 0 0 0 0-8Zm-5 9a2 2 0 0 0-2 2c0 1.7.83 2.97 2.13 3.8A9.14 9.14 0 0 0 9 18c.95 0 1.88-.1 2.72-.31l-1.83-2a3.4 3.4 0 0 1 .1-4.69H4Zm10.5 1-.63-.49a2.4 2.4 0 0 0-3.24 3.5l3.5 3.83c.2.22.54.22.74 0l3.5-3.83a2.4 2.4 0 0 0-3.24-3.5l-.63.5Z"
              fill="currentColor"
            />
          </svg>
        </span>
        Mine
      </button>
      <button
        aria-expanded={false}
        aria-labelledby="splitButton-r2__primaryActionButton"
        className="fui-Button rsawnvh fui-MenuButton fui-SplitButton__menuButton ___1qj6wbq_1bdujhx f1sbtcvk fwiuce9 fdghr9 f15vdbe4 fwbmr0d f44c6la ftqa4ok f2hkw1w frrbwxo f1um7c6d f6pwzcr"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <span
          className="fui-MenuButton__menuIcon ___1gqzbv5_ogidrh0 fwrc4pm f1ugzwwg fvblgha frx94fk"
        >
          <svg
            aria-hidden={true}
            className="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
            fill="currentColor"
            height="1em"
            viewBox="0 0 20 20"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </header>
  <main
    className="___1qdybx0_0000000 fqerorx f5ogflp fzkkow9 f68mrw8 fihm6t0"
  >
    <div
      className="fui-Spinner ___p104zz0_oywd6f0 f22iagw f122n59 f4d9j23 fez10in f4px1ci fn67r4l f1063pyq"
      role="progressbar"
    >
      <span
        className="fui-Spinner__spinner ___1a7f5ix_1qxa57p f1or16p5 f1grzc83 f19shzzi f5tbecn fn4mtlg f1y80fxs f1r2crtq f1wsi8sr f1bkm2qd f1urqz7h f1da2vov f11rfva0 f1exc66 f89rf2a f1w2xg3q f1flujpd fmn74v6 f11ditju f1m9nikz fq8a5sv f1b4lwqj f1najlst fjxod4 fu3xdw0 f1ttdh6v f1eseayc folzdkc fhlfkde f61h2gu flglbw1"
      >
        <svg
          className="fui-Spinner__Progressbar"
        >
          <circle
            className="fui-Spinner__Track"
          />
          <circle
            className="fui-Spinner__Tail"
          />
        </svg>
      </span>
    </div>
  </main>
</div>
`;
